forage-android/src/main/java/com/joinforage/forage/android/ecom/services/vault/bt/BTResponseParser.kt

```diff
@@ -0,0 +1,63 @@
+package com.joinforage.forage.android.ecom.services.vault.bt
+
+import com.joinforage.forage.android.core.services.forageapi.network.ForageApiError
+import com.joinforage.forage.android.core.services.forageapi.network.ForageApiResponse
+import com.joinforage.forage.android.core.services.forageapi.network.ForageError
+import com.joinforage.forage.android.core.services.forageapi.network.UnknownErrorApiResponse
+import com.joinforage.forage.android.core.services.vault.VaultResponseParser
+
+class BTResponseParser(btRes: Result<Any?>) : VaultResponseParser {
+    override val isNullResponse: Boolean = false
+    override val vaultErrorMsg: String = btRes.exceptionOrNull().toString()
+    override val rawResponse: String = btRes.toString()
+
+    override val vaultError: ForageApiResponse.Failure?
+    override val forageError: ForageApiResponse.Failure?
+    override val successfulResponse: ForageApiResponse.Success<String>?
+
+    val isSuccessful: Boolean
+
+    init {
+        // BT returns a Result<Any?> so it's never null
+        vaultError = parseVaultError(btRes)
+        forageError = parseForageError(BaseResponseRegExp(btRes))
+        isSuccessful = btRes.isSuccess && vaultError == null && forageError == null
+        successfulResponse = parseSuccessfulResponse(btRes)
+    }
+
+    private fun parseVaultError(vaultResponse: BasisTheoryResponse): ForageApiResponse.Failure? {
+        if (vaultResponse.isSuccess) return null
+        val resRegExp = BtResponseRegExp(vaultResponse)
+        return try {
+            // TODO: add DD metric to track frequency of proxy errors
```

Comment:
The caller captures proxy errors using the vault monitor/measurer `[VaultProxyResponseMonitor]`. Can remove or update this comment to include the context from this comment

Reply:
Ah nice! OK I'll modify the comment to capture this. Thanks!

Reply:
Updated the comment!